openmdao/components/interp_util/tests/test_interp_nd.py

```diff
@@ -161,8 +162,180 @@ def test_cs_across_interp(self):
+                    np.cos(u * v),
+                    -u**2 * np.sin(u * v) + u * v * np.cos(u * v) +
+                    np.sin(u * v))
+
+        # uniformly spaced axis
+        u = np.linspace(0, 3, 50)
+        # randomly spaced axis
+        np.random.seed(7590)
+        v = np.random.uniform(0, 3, 50)
+        v.sort()
+
+        points = [u, v]
+        values = f(*np.meshgrid(*points, indexing='ij'))
+        return points, values, f, df
+
+    def _get_sample_4d_large(self):
+        def f(x, y, z, w):
+            return x**2 + y**2 + z**2 + w**2
+        X = np.linspace(-10, 10, 6)
+        Y = np.linspace(-10, 10, 7)
+        np.random.seed(0)
+        Z = np.random.uniform(-10, 10, 6)
+        Z.sort()
+        W = np.linspace(-10, 10, 8)
+        points = [X, Y, Z, W]
+        values = f(*np.meshgrid(*points, indexing='ij'))
+        return points, values
+
+    def test_2d(self):
+        # test interpolated values
+        points, values, func, df = self._get_sample_2d()
+        np.random.seed(1)
+        X, Y = np.meshgrid(*points, indexing='ij')
+        X = X.ravel()
+        Y = Y.ravel()
+        grid = np.array([X, Y]).T
+
+        test_pt = np.random.uniform(0, 3, 2)
+        actual = func(*test_pt)
+        for method in self.interp_methods:
+            interp = InterpNDSemi(grid, values.ravel(), method=method)
+            computed = interp.interpolate(test_pt)
+            r_err = rel_error(actual, computed)
+            assert r_err < self.tol[method]
+
+    def test_minimum_required_gridsize(self):
+        for method in self.interp_methods:
+
+            k = self.interp_configs[method] - 1
+            x = np.linspace(0, 1, k)
+            y = np.linspace(0, 1, k)
+
+            points = [x, y]
+            X, Y = np.meshgrid(*points, indexing='ij')
+            X = X.ravel()
+            Y = Y.ravel()
+            values = X + Y
+            grid = np.array([X, Y]).T
+            with self.assertRaises(ValueError) as cm:
+                interp = InterpNDSemi(grid, values, method=method)
+
+            msg = 'There are {} points in a data dimension, but method'.format(k)
+            self.assertTrue(str(cm.exception).startswith(msg))
+
+    def test_NaN_exception(self):
+        np.random.seed(1234)
+        x = np.linspace(0, 2, 5)
+        y = np.linspace(0, 1, 7)
+        points = [x, y]
+        X, Y = np.meshgrid(*points, indexing='ij')
+        X = X.ravel()
+        Y = Y.ravel()
+        grid = np.array([X, Y]).T
+        values = np.random.rand(5, 7).ravel()
+        interp = InterpNDSemi(grid, values, method='slinear', extrapolate=False)
+
+        with self.assertRaises(OutOfBoundsError) as cm:
+            interp.interpolate(np.array([1, np.nan]))
+
+        err = cm.exception
+
+        self.assertEqual(str(err), 'One of the requested xi contains a NaN')
+        self.assertEqual(err.idx, 1)
+        self.assertTrue(np.isnan(err.value))
+
+    def ztest_error_messages(self):
```

Comment:
why the z here?

Reply:
Oops, I was commenting out some tests when I was running in the debugger. Forgot to uncomment it.